device_attributes/index.bs

```diff
@@ -0,0 +1,192 @@
+ 
+The API is presumed to be used by web applications, which are understood and managed by the device administrators.
+ 
+## Device attributes ## {#device-attributes}
+ 
+The API is presumed to be used with a management infrastructure which allows administrators to manage a fleet of devices and configure properties for those devices.
+ 
+### <dfn>Annotated Asset ID</dfn> ### {#annotated-asset-id-section}
+
+An administrator-defined value which uniquely identifies a device within an organization.
+
+### <dfn>Annotated Location</dfn> ### {#annotated-location-section}
+
+An administrator-defined value which uniquely identifies a location within an organization.
+
+### <dfn>Directory ID</dfn> ### {#directory-id-section}
+ 
+An inventory management system-defined value which uniquely identifies a device within an organization.
+ 
+### <dfn>Hostname</dfn> ### {#hostname-section}
+ 
+An administrator-defined value which is used as the device hostname during DHCP requests.
+ 
+### <dfn>Serial Number</dfn> ### {#serial-number-section}
+ 
+A manufacturer-defined value which uniquely identifies a device among those produced by that manufacturer.
+ 
+# {{NavigatorManagedData}} interface # {#navigatormanageddata-interface}
+ 
+<xmp class="idl">
+[
+  SecureContext,
+  Exposed=Window
+] interface NavigatorManagedData : EventTarget {
+  // Device Attributes API.
+  Promise<DOMString> getAnnotatedAssetId();
+  Promise<DOMString> getAnnotatedLocation();
+  Promise<DOMString> getDirectoryId();
+  Promise<DOMString> getHostname();
+  Promise<DOMString> getSerialNumber();
+};
+</xmp>
+ 
+Methods on this interface typically complete asynchronously, queuing work on the <dfn>managed data task source</dfn>.
+ 
+## {{getAnnotatedAssetId()}} method ## {#getAnnotatedAssetId-method}
+ 
+  The <dfn method for="NavigatorManagedData">getAnnotatedAssetId()</dfn> method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. If this API is not called by a managed web application with permission to access device attributes, reject |promise| with a {{NotAllowedError}} DOMException and abort these steps.
+      1. If no [=Annotated Asset ID=] has been set by the administrator, resolve |promise| with {{undefined}} value and abort these steps.
+      1. [=Queue a global task=] on the [=relevant global object=] of [=this=] using the [=managed data task source=] to [=resolve=] promise with the [=Annotated Asset ID=].
+  1. Return |promise|.
+ 
+## {{getAnnotatedLocation()}} method ## {#getAnnotatedLocation-method}
+ 
+  The <dfn method for="NavigatorManagedData">getAnnotatedLocation()</dfn> method steps are:
+  1. Let |promise| be [=a new promise=].
+  1. Run the following steps in parallel:
+      1. If this API is not called by a managed web application with permission to access device attributes, reject |promise| with a {{NotAllowedError}} DOMException and abort these steps.
```

Comment:
These steps are still missing the "queue a task" part of resolving or rejecting _promise_.

Reply:
Hmm, I reprased the description of these methods accordingly, but don't have confidence whether they are correct now. Please help to review them again. Thanks!